nir/src/main/scala/scala/scalanative/nir/Vals.scala

```diff
@@ -175,59 +175,7 @@ object Val {
   final case class StructValue(values: Seq[Val])                  extends Val
   final case class ArrayValue(elemty: nir.Type, values: Seq[Val]) extends Val
   final case class Chars(value: java.lang.String) extends Val {
```

Comment:
Since `c""` strings are `CString`s, which are `Ptr[CChar] = Ptr[Byte]`, we should directly store `value` as a `Seq[Byte]` at this point. The compiler plugin will then have to encode a `String` as a sequence of `Byte`s using UTF-8.